peekingduck/declarative_loader.py

```diff
@@ -180,12 +178,69 @@ def _edit_config(
                         f"Config for node {node_name} does not have the key: {key}"
                     )
                 else:
+                    if key == "detect_ids" and node_name in (
+                        "model.yolo",
+                        "model.efficientdet",
+                    ):
+                        key, value = self._change_labels_to_ids(node_name, key, value)
                     dict_orig[key] = value
                     self.logger.info(
                         f"Config for node {node_name} is updated to: '{key}': {value}"
                     )
         return dict_orig
 
+    def _change_labels_to_ids(
+        self, node_name: str, key: str, value: list
+    ) -> Tuple[str, list]:
+        """Process model.yolo or model.efficientdet detect_ids and check for labels to
+        be converted to ids, e.g. person to 0, car to 2
+
+        Args:
+            node_name (str): to determine if node is yolo or efficientdet as both
+                             lists of ids are different.
+            key (str): can only be "detect_ids", else error.
+            value (list): list of labels/ids for detection.
```

Comment:
If we are certain enough that this function only handles converting labels to IDs, perhaps can make the variable names more explicit? Maybe something like `labels_and_ids`

Reply:
`_change_labels_to_ids` is a function that maps:
1. string to int (1-to-1)
2. int to int (ID)

As the latter is essentially an identity function, I think it's ok to name it as such. It is verbose to call it `_change_labels_and_ids_to_ids`.

Reply:
I was referring the `value` argument/variable, thinking that naming it to `labels_and_ids` might make `for x in labels_and_ids` in the list comprehension more descriptive. But i guess the current naming is fine since that line of code is pretty close to the documentation

Reply:
Oic. The name `value` is meant to reflect the `key`/`value` pair of the node attribute (specifically, `key=detect_ids`, `value=list of ids/labels`), and these two are returned to caller `DeclarativeLoader` to replace the current `key`/`value` pair being processed under [`_edit_config`](https://github.com/ongtw/PeekingDuck/blob/465312febe0b4ca4b9a898225a238690ab414647/peekingduck/declarative_loader.py#L185).